### DISCUSSION TITLE
How do we migrate custom product options and tab to the new product editor?

### DISCUSSION THREAD

**Discussion by rogercoathup** [woocommerce/woocommerce#47329] [WooCommerce New Product Editor]
With the current product editor, it's a simple process to add custom product options to the editor.  We'd hook e.g. on 
`woocommerce_product_options_general_product_data` to add options to the general tab, and hook on `woocommerce_process_product_meta` to save the options.

It was similarly simple to add new tabs (and indeed is the approach of a number of plugins e.g. dynamic pricing, facebook for wp, gravity forms, affiliateWP).

How do we add & save custom product options with the new product editor?

**Comment by louwie17** [comment]
Hi @rogercoathup, thanks for the question!
It is also simple to add new fields or tabs into the new editor, we do have two blog posts with examples around this ( more to come ):
- Show casing how to use our generic blocks that support meta data: https://developer.woocommerce.com/2023/11/17/getting-to-know-woo-extensibility-in-the-new-product-editor/
- How to write a more custom field: https://developer.woocommerce.com/2024/01/17/getting-to-know-woo-extending-the-new-product-editor-with-react/ 

I also recommend looking at our handbook: https://developer.woocommerce.com/docs/product-editor-development-handbook/

Let us know if you have further questions?
